syntax = "proto3";

message AddCommentRequest {
    int32 user_id = 1;
    int32 song_id = 2;
    string comment = 3;
}

message AddCommentResponse {
    int32 response = 1;
}

message RemoveCommentRequest {
    int32 user_id = 1;
    int32 song_id = 2;
    int32 comment_id = 3;
}

message RemoveCommentResponse {
    int32 response = 1;
}

message UpdateCommentRequest {
    int32 user_id = 1;
    int32 song_id = 2;
    int32 comment_id = 3;
    string comment = 4;
}

message UpdateCommentResponse {
    int32 response = 1;
}

service CommentService {
    rpc Add (AddCommentRequest) returns (AddCommentResponse);
    rpc Update (UpdateCommentRequest) returns (UpdateCommentResponse);
    rpc Remove (RemoveCommentRequest) returns (RemoveCommentResponse);
}